### ISSUE TITLE
Selecting on bitslice in parser crashes bmv2

### ISSUE THREAD

**Issue by pierce-m** [p4lang/p4c#387]
This is because bit slices get lowered to shift expressions which bmv2 can't evaluate in transition keys: example below.
```
#include <core.p4>
#include <v1model.p4>

header A {
    bit<3> f1;
    bit<1> bos;
}

struct H {
    A a_bos;
    A[3] a;
}

struct M {
}

parser p(packet_in packet, out H hdr, inout M meta, inout standard_metadata_t sm) {
    state start {
        transition parse_A;
    }
    state parse_A {
        transition select(packet.lookahead<bit<4>>()[3:3]) {
            1w0 : parse_A_not_bos;
            1w1 : parse_A_bos;
        }
    }
    state parse_A_not_bos {
        packet.extract(hdr.a.next);
        transition parse_A;
    }
    state parse_A_bos {
        packet.extract(hdr.a_bos);
        transition accept;
    }
}

control verifyCk(in H hdr, inout M meta) {
    apply { }
}

control ingress(inout H hdr, inout M meta, inout standard_metadata_t sm) {
    apply { }
}

control egress(inout H hdr, inout M meta, inout standard_metadata_t sm) {
    apply { }
}

control computeCk(inout H hdr, inout M meta) {
    apply { }
}

control deparser(packet_out b, in H hdr) {
    apply {
        b.emit(hdr.a);
        b.emit(hdr.a_bos);
    }
}

V1Switch(p(), verifyCk(), ingress(), egress(), computeCk(), deparser()) main;

```

**Comment by mbudiu-vmw** [comment]
The outstanding PR #367 should fix this issue once it is merged.

**Comment by mbudiu-vmw** [comment]
This is a duplicate of https://github.com/p4lang/p4c/issues/356

**Comment by pierce-m** [comment]
Thanks, I didn't see #356
